### PR TITLE
Provide option for explicit toString calls in the generated interpolation code

### PR DESCRIPTION
**Pull request by PawelLipski**
I found it pretty useful e.g. for static code analyzers to make sure that an explicit `toString` call is always present. E.g. in our project we'd like to forbid `toString` calls on certain classes (like `Option(al)`), since it's usually not intended (and in case of `Option`, it can result in a literal `Some(...)` being rendered in user-facing strings in the UI).

Plus also resolve a bug related to reported positions of compilation errors within interpolated strings & improve code style in a few places.

**Depends on https://github.com/jupiter-tools/compile-test/pull/1**

### CODE REVIEW

File: src/main/java/com/antkorwin/betterstrings/BetterStringsProcessor.java [hunk 1]
```diff
@@ -71,9 +74,4 @@ private boolean isClassOrEnum(Element codeElement) {
 	public SourceVersion getSupportedSourceVersion() {
 		return SourceVersion.latestSupported();
 	}
-
-	private void printBanner() {
-		String banner = "v0.3 String Interpolation Java Plugin, by Anatoliy Korovin";
```
Comment (PawelLipski) [inline src/main/java/com/antkorwin/betterstrings/BetterStringsProcessor.java:76]: Resolving #7 btw ;)

File: src/main/java/com/antkorwin/betterstrings/ast/ExpressionParser.java [hunk 2]
```diff
@@ -45,24 +45,21 @@ private CompilationUnitTree getCompilationUnitTree(String code) {
...
 	private class FakeJavaFileWrapper extends SimpleJavaFileObject {
 
-		private String text;
+		private final String text;
 
 		public FakeJavaFileWrapper(String text) {
 			super(URI.create("myfake:/Test.java"), JavaFileObject.Kind.SOURCE);
-			this.text = "class Test { Object value = String.valueOf(" + text + "); }";
+			if (callToStringExplicitlyInInterpolations) {
+				this.text = "class Test { Object value = java.util.Objects.nonNull(" + text + ") ? java.util.Objects.requireNonNull(" + text + ").toString() : \"null\"; }";
```
Comment (PawelLipski) [inline src/main/java/com/antkorwin/betterstrings/ast/ExpressionParser.java:59]: That's the best I've come up with so far, it performs boxing for primitive types, but unfortunately requires double evaluation of the expression :/

**Comment by antkorwin** [comment]
Hi @PawelLipski ,
thanks a lot for your work!
it looks pretty useful!

I will find time to cover this by unit tests and fix build errors, 
unfortunately, now I'm very busy (a lot of work this week), maybe I can do it next week.

**Comment by PawelLipski** [comment]
Ok! let me know if you agree with the general design, I can also provide docs for this as well

**Comment by PawelLipski** [comment]
Also, I fixed the build (see the latest commit), apparently `com.sun.tools.javac.api.JavacTaskImpl#parse` is declared throwing or not throwing `IOException` depending on the exact environment (Java version? I develop on OpenJDK/JBR 11.0.6)

**Comment by PawelLipski** [comment]
@antkorwin added tests for the newly-added option; they depends on https://github.com/jupiter-tools/compile-test/pull/1 getting released as v0.2 first, however

**Comment by PawelLipski** [comment]
... and docs as well :) I'd appreciate your feedback since one of our projects would greatly benefit from this feature getting released

**Comment by mkondratek** [comment]
Hey @antkorwin :smile:  how does the work go? when can we get it merged?

File: pom.xml [hunk 3]
```diff
@@ -147,7 +147,7 @@
         <dependency>
             <groupId>com.jupiter-tools</groupId>
             <artifactId>compile-test</artifactId>
-            <version>0.1</version>
+            <version>0.2</version>
```
Comment (PawelLipski) [inline pom.xml:150]: @antkorwin so now that compile-test v0.2 is on Maven Central (:tada:), could you re-run the most recent test? I apparently can't trigger a test re-run on Travis for this project myself. Or I can just amend & force-push the most recent commit.

**Comment by PawelLipski** [comment]
@antkorwin long time no hear... how's going?

**Comment by antkorwin** [comment]
Hi, @PawelLipski 
I'm so sorry about long delay

It looks great!
Thanks a lot for your patience!

**Comment by antkorwin** [comment]
I will check it today and deploy the next version to maven central...

@PawelLipski  @mkondratek  thanks a lot for your contribution!

File: README.adoc [hunk 4]
```diff
@@ -132,14 +137,43 @@ System.out.println("${'$'}{a+b}");
...
+----
+
+will yield:
+
+[source, java]
+----
+"Result: "
+  + String.valueOf(obj)
+  + ".method() = "
+  = String.valueOf(obj.method())
```
Comment (PawelLipski) [inline README.adoc:160]: Instead of `=` it should be `+` here... to be fixed with the next release

**Comment by antkorwin** [comment]
the next release already published to the maven central
https://github.com/antkorwin/better-strings/releases/tag/v0.4

File: README.adoc (hunk 4 above)
Comment (antkorwin) [reply README.adoc:160]: wow, I read this text but didn't notice a typo =)
I have fixed it now.

**Pull request merged**